oshi-core/src/main/java/oshi/software/os/linux/LinuxOperatingSystem.java

```diff
@@ -256,18 +265,19 @@ public OSProcess getProcess(int pid) {
         // See man proc for how to parse /proc/[pid]/io
         proc.setBytesRead(ParseUtil.parseLongOrDefault(MapUtil.getOrDefault(io, "read_bytes", ""), 0L));
         proc.setBytesWritten(ParseUtil.parseLongOrDefault(MapUtil.getOrDefault(io, "write_bytes", ""), 0L));
-        // The stat structure on Linux does not have consistent ordering or byte
-        // size accross architectures so we are forced to use the stat command
-        List<String> stat = ExecutingCommand.runNative("stat -c %u,%U,%g,%G /proc/" + pid);
-        if (!stat.isEmpty()) {
-            split = stat.get(0).split(",");
-            if (split.length == 4) {
-                proc.setUserID(split[0]);
-                proc.setUser(split[1]);
-                proc.setGroupID(split[2]);
-                proc.setGroup(split[3]);
-            }
+        
+        Map<String, String> status = FileUtil.getKeyValueMapFromFile(String.format("/proc/%d/status", pid), ":");
+        proc.setUserID(ParseUtil.whitespaces.split(MapUtil.getOrDefault(status, "Uid", ""))[0]);
+        proc.setGroupID(ParseUtil.whitespaces.split(MapUtil.getOrDefault(status, "Gid", ""))[0]);
+        if (users == null) {

```

Comment:
In the single-PID case, both `users` and `groups` would be `null` and you would end up calling `getUserInfo()` and incurring the object creation overhead twice.  Call it once (if either variable is `null`) similar to how you did in  LinuxOperatingSystem line 195.  Or better yet, don't call this method with null, and call `getUserInfo()` before calling this method.